stable/datadog/values.yaml

```diff
@@ -1,290 +1,256 @@
+## @param image - object - required
+## Define the Datadog image to work with.
 #
-# HINT: If you want to use datadog.collectEvents, keep deployment.replicas set to 1.
-deployment:
-  enabled: false
-  replicas: 1
-  # Affinity for pod assignment
-  # Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
-  affinity: {}
-  # Tolerations for pod assignment
-  # Ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
-  tolerations: []
-  # If you're using a NodePort-type service and need a fixed port, set this parameter.
-  # dogstatsdNodePort: 8125
-  # traceNodePort: 8126
-
-  service:
-    type: ClusterIP
-    annotations: {}
-
-  ## Sets PriorityClassName if defined
-  # priorityClassName:
-
-## deploy the kube-state-metrics deployment
-## ref: https://github.com/kubernetes/charts/tree/master/stable/kube-state-metrics
-
-kubeStateMetrics:
-  enabled: true
+image:
 
-# This is the new cluster agent implementation that handles cluster-wide
-# metrics more cleanly, separates concerns for better rbac, and implements
-# the external metrics API so you can autoscale HPAs based on datadog
-# metrics
-clusterAgent:
-  containerName: cluster-agent
-  image:
-    repository: datadog/cluster-agent
-    tag: 1.1.0
-    pullPolicy: IfNotPresent
-  enabled: false
-  ## This needs to be at least 32 characters a-zA-z
-  ## It is a preshared key between the node agents and the cluster agent
-  token: ""
-  replicas: 1
-  ## Enable the metricsProvider to be able to scale based on metrics in Datadog
-  metricsProvider:
-    enabled: false
-  resources:
-    requests:
-      cpu: 200m
-      memory: 256Mi
-    limits:
-      cpu: 200m
-      memory: 256Mi
+  ## @param repository - string - required
+  ## Define the repository to use:
+  ## use "datadog/agent" for Datadog Agent 6 
+  ## use "datadog/dogstatsd" for Standalone Datadog Agent DogStatsD6
+  #
+  repository: datadog/agent
+
+  ## @param tag - string - required
+  ## Define the Agent version to use.
+  ## Use 6.9.0-jmx to enable jmx fetch collection
+  # 
+  tag: 6.9.0
+
+  ## @param pullPolicy - string - required
+  ## The Kubernetes pull policy.
+  #
+  pullPolicy: IfNotPresent
 
-  ## Override the agent's liveness probe logic from the default:
-  ## In case of issues with the probe, you can disable it with the
-  ## following values, to allow easier investigating:
-  # livenessProbe:
-  #   exec:
-  #     command: ["/bin/true"]
-  ## Override the cluster-agent's readiness probe logic from the default:
-  # readinessProbe:
+  ## @param pullSecrets - list of key:value strings - optional
+  ## It is possible to specify docker registry credentials
+  ## See https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod
+  #
+  #  pullSecrets:
+  #    - name: "<REG_SECRET>"
 
 datadog:
-  ## You'll need to set this to your Datadog API key before the agent will run.
+
+  ## @param apiKey - string - required
```

Comment:
(apiKey OR apiKeyExistingSecret) is required, should we group them? the latter takes precendence btw:
  - if apiKeyExistingSecret is set, we use this secret name as DD_API_KEY
  - else, we create a secret with the contents of apiKey and use it

Same logic for appKey and appKeyExistingSecret